### PR TITLE
test: add fuzz test for metric region migration

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Add fuzz test for metric region migration

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

**Review by fengjiachun — APPROVED** [review]
LGTM

File: tests-fuzz/targets/migration/fuzz_migrate_metric_regions.rs [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+        .collect::<Vec<_>>();
+    info!("List datanodes: {:?}", datanodes);
+
+    // Generates region migration task.
+    let mut migrations = Vec::with_capacity(num_partitions);
+    let mut new_distribution: BTreeMap<u64, HashSet<_>> = BTreeMap::new();
+    for (datanode_id, regions) in region_distribution {
+        let step = rng.gen_range(1..datanodes.len());
+        for region in regions {
+            let to_peer = (datanode_id + step as u64) % datanodes.len() as u64;
```
Comment (evenyag) [inline tests-fuzz/targets/migration/fuzz_migrate_metric_regions.rs:253]: Is it possible that the `to_peer` is the same as the current datanode id?

File: tests-fuzz/targets/migration/fuzz_migrate_metric_regions.rs [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+            |partition| {
+                info!("Region: {region_id},  datanode: {}", partition.datanode_id);
+                partition.datanode_id == migration.to_peer
+            },
+            Duration::from_secs(1),
+        )
+        .await;
+    }
+
+    // Creates more logical tables and inserts values
```
Comment (evenyag) [inline tests-fuzz/targets/migration/fuzz_migrate_metric_regions.rs:415]: Why do we need to repeat this so many times?

File: tests-fuzz/targets/migration/fuzz_migrate_metric_regions.rs (hunk 1 above)
Comment (WenyXu) [reply tests-fuzz/targets/migration/fuzz_migrate_metric_regions.rs:253]: Nope

**Review by evenyag — APPROVED** [review]

**Pull request merged**